### PR TITLE
feat: init samples and packages

### PR DESCRIPTION
**Pull request by lslzl3000**
init basic sample index, load all demos in /samples
init packages:
- internal extenstions: stats, physics, media, post
- external libs: ammo, draco

### CODE REVIEW

File: packages/physics/Physics.ts
Comment (lslzl3000) [inline packages/physics/Physics.ts:1]: need update docs
and test with new engine life cycle

File: packages/physics/Rigidbody.ts
Comment (lslzl3000) [inline packages/physics/Rigidbody.ts:1]: 1. translate
2. test with new engine

File: packages/media-extention/ChromaKeyMaterial.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
+import { Engine3D, MaterialBase, ShaderLib, Vector4, Color, BlendMode, registerMaterial } from "@orillusion/core";
+import ChromaKeyShader from "./ChromaKeyShader.wgsl?raw";
+
+/**
+ * Video 材质(ChromaKey)
+ * 不计算光照，仅通过Video像素颜色渲染的基础材质，并且过滤了背景色.
```
Comment (hellmor) [inline packages/media-extention/ChromaKeyMaterial.ts:6]: Language conversion required

File: packages/media-extention/ChromaKeyMaterial.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ * @group Material
+ */
+export class ChromaKeyMaterial extends MaterialBase {
+
+        /**
+         * 创建新的Video材质对象
```
Comment (hellmor) [inline packages/media-extention/ChromaKeyMaterial.ts:12]: Language conversion required

File: packages/media-extention/ChromaKeyMaterial.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        public set rectClip(value: Vector4) {
+                this.renderShader.uniforms[`rectClip`].vector4 = value;
+        }
+
+        public get rectClip(): Vector4 {
+                return this.renderShader.uniforms[`rectClip`].vector4;
+        }
+
+        /**
+         * 背景关键色
```
Comment (hellmor) [inline packages/media-extention/ChromaKeyMaterial.ts:56]: Language conversion required.
More is needed below.

File: packages/media-extention/ImageMaterial.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import { Engine3D, MaterialBase, ShaderLib, Vector4, Color, Texture } from "@orillusion/core";
+import ImageMaterialShader from "./ImageMaterialShader.wgsl?raw";
+
+
+/**
+ * ImageMaterial
+ * 不计算光照，仅通过Image像素颜色渲染的基础材质
```
Comment (hellmor) [inline packages/media-extention/ImageMaterial.ts:7]: Language conversion required

File: packages/media-extention/ImageMaterial.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+ * @group Material
+ */
+export class ImageMaterial extends MaterialBase {
+
+    /**
+     * 创建新的Video材质对象
```
Comment (hellmor) [inline packages/media-extention/ImageMaterial.ts:13]: Language conversion required

File: packages/media-extention/ImageMaterial.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    public set rectClip(value: Vector4) {
+        this.renderShader.uniforms[`rectClip`].vector4 = value;
+    }
+
+    public get rectClip(): Vector4 {
+        return this.renderShader.uniforms[`rectClip`].vector4;
+    }
+
+    /**
+     * 设置材质环境贴图
```
Comment (hellmor) [inline packages/media-extention/ImageMaterial.ts:47]: Language conversion required

File: packages/media-extention/VideoMaterial.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
+
+import { Color, Engine3D, MaterialBase, ShaderLib, Texture, Vector4, registerMaterial } from '@orillusion/core';
+import VideoShader from './VideoShader.wgsl?raw'
+
+/**
+ * Video 材质
+ * 不计算光照，仅通过Video像素颜色渲染的基础材质
```
Comment (hellmor) [inline packages/media-extention/VideoMaterial.ts:7]: Language conversion required

File: packages/media-extention/VideoTexture.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        if (old) {
+            old.pause()
+            old.src = old.srcObject = null
+            old.load()
+        }
+    }
+
+    /**
+     * 更新gpu纹理
```
Comment (hellmor) [inline packages/media-extention/VideoTexture.ts:44]: Language conversion required

File: packages/physics/HingeConstraint.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+    public get targetRigidbody(): Rigidbody {
+        return this._targetRigidbody;
+    }
+
+    public set targetRigidbody(value: Rigidbody) {
+        this._targetRigidbody = value;
+    }
+
+    // 初始化btRigidbody
```
Comment (hellmor) [inline packages/physics/HingeConstraint.ts:66]: Language conversion required

File: packages/physics/Physics.ts [hunk 10]
```diff
@@ -0,0 +1,181 @@
+import Ammo from '@orillusion/ammo';
+import {BoundingBox, Vector3, Time} from '@orillusion/core'
+import { Rigidbody } from './Rigidbody';
+
+/**
+ * 物理引擎
```
Comment (hellmor) [inline packages/physics/Physics.ts:6]: Language conversion required

File: packages/physics/Physics.ts [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    private __updateWithDelta(time:number ) {
+        if(!this._isInited) return;
+        if(this.isStop) return;
+        this._world.stepSimulation(time);
+    }
+
+    private __updateWithFixedTimeStep(time:number, maxSubSteps:number, fixedTimeStep:number ) {
+        if(!this._isInited) return;
```
Comment (hellmor) [inline packages/physics/Physics.ts:131]: Need to wrap

File: packages/physics/Physics.ts [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+        this._isInited = true;
+        this.physicBound = new BoundingBox(new Vector3(), new Vector3(2000, 2000, 2000));
+    }
+
+    public get isInited(): boolean {
+        return this._isInited;
+    }
+
+    private __updateWithDelta(time:number ) {
+        if(!this._isInited) return;
```
Comment (hellmor) [inline packages/physics/Physics.ts:125]: Need to wrap

File: packages/physics/Physics.ts [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+        if(this.isStop) return;
+        this.world.stepSimulation(time, maxSubSteps, fixedTimeStep);
+    }
+
+
+    public update() {
+        if (!this._isInited) {
+            return;
+        }
+        if (this.isStop) return;
```
Comment (hellmor) [inline packages/physics/Physics.ts:141]: Need to wrap

File: packages/physics/Physics.ts [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+                body.btRigidbody.activate(false);
+                // this._world.removeRigidBody(body.btRigidbody);
+                body.destroy();
+            }
+        }
+    }
+}
+
+/**
+ * 物理引擎实例，通常全局只申明一个实例
```
Comment (hellmor) [inline packages/physics/Physics.ts:171]: language

File: packages/physics/Rigidbody.ts [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+        this._friction = value;
+        if (this._btRigidbody) this._btRigidbody.setFriction(value);
+    }
+    public get rollingFriction(): number {
+        return this._rollingFriction;
+    }
+
+    public set rollingFriction(value: number) {
+        this._rollingFriction = value;
+        if (this._btRigidbody) this._btRigidbody.setRollingFriction(value);
```
Comment (hellmor) [inline packages/physics/Rigidbody.ts:77]: Need to wrap

File: packages/physics/Rigidbody.ts [hunk 16]
```diff
@@ -0,0 +1,264 @@
...
+    }
+
+    public get restitution(): number {
+        return this._restitution;
+    }
+    public set restitution(value: number) {
+        this._restitution = value;
+        if (this._btRigidbody) this._btRigidbody.setRestitution(value);
```
Comment (hellmor) [inline packages/physics/Rigidbody.ts:85]: Need to wrap

File: packages/physics/Rigidbody.ts [hunk 17]
```diff
@@ -0,0 +1,264 @@
...
+    }
+
+    /**
+     * 组件是否初始化
```
Comment (hellmor) [inline packages/physics/Rigidbody.ts:89]: language

**Review by Codeboy-cn — APPROVED** [review]

**Pull request merged**